### PR TITLE
Currrency: Swapped API versions

### PR DESCRIPTION
**Pull request by pjhampton**
# PLEASE DON'T MERGE ME - WIP 🎸 

## Description of new Instant Answer, or changes

This swaps out the old XE API with the new one. **The end user shouldn't notice a difference**. 

## People to notify

@moollaza 

---

Instant Answer Page: https://duck.co/ia/view/currency

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: share/spice/currency/currency.js [hunk 1]
```diff
@@ -286,14 +278,17 @@
...
             }
 
-            var endpoint = "/js/spice/currency/1/" + from + "/" + to;
-            $.get(endpoint, function(payload) {
-
-                // jsonp is returned from the API so we have to alter the contents
+            $.ajax({
+                url: "/js/spice/currency/1/" + from + "/" + to,
+                beforeSend: function( xhr ) {
+                    xhr.overrideMimeType( "text/plain; charset=x-user-defined" );
```
Comment (moollaza) [inline share/spice/currency/currency.js]: What is this for?

File: share/spice/currency/currency.js [hunk 2]
```diff
@@ -286,14 +278,17 @@
...
+                },
+                error: function(request, status, error) {
+                    Spice.failed('currency');
+                }
+            }).done(function(payload) {
                 var response = JSON.parse(payload.trim().replace(/^[^\(]*\(/, '').replace(/\);$/, ''));
```
Comment (moollaza) [inline share/spice/currency/currency.js]: Perhaps we should just make an `alt_to` in the Perl so we don't need to parse the response via regex -- that seems like a much safer solution.

File: share/spice/currency/currency.js (hunk 1 above)
Comment (moollaza) [inline share/spice/currency/currency.js]: What's this for?

Comment (pjhampton) [inline share/spice/currency/currency.js]: Both your comments are related. The api is returning a jsonp response which for one reason or another is throwing an error each time I run `$.ajax`, `$.getJSON` so I'm returning it as text and parsing into valid json. In the above line I'm overriding the MIME type of the response to plain text.

Comment (moollaza) [inline share/spice/currency/currency.js]: Ah, I see. I think avoiding these work around with a new endpoint is a simpler solution. We can make alt to called `currency_json` and disable the jsonp wrapping.

Then you can `$.getJSON('/js/spice/currency_json/' ... `

**Comment by moollaza** [comment]
Closing as it appears this won't be moving forward :(

**Comment by pjhampton** [comment]
No problemo @moollaza 👌  onwards and upwards :)

**Pull request closed without merging**